### PR TITLE
Bulk delete

### LINKED ISSUE #2113: Moderation Bulk Delete
Script to bulk delete spam moderation requests

### PR DESCRIPTION
**Pull request by dakota002**
Resolves #2113

None selected:

image

Some selected:
image

### CODE REVIEW

File: app/routes/circulars.moderation._index.tsx [hunk 1]
```diff
@@ -31,45 +39,100 @@ export async function loader({ request }: LoaderFunctionArgs) {
...
-          />
-        ))}
-      </SegmentedCards>
+      <Form method="POST">
+        <ButtonGroup>
+          <Link to="/circulars" className="usa-button usa-button--outline">
+            Back
+          </Link>
+          <Button type="submit" secondary disabled={disableSubmit}>
+            Bulk Delete
```
Comment (lpsinger) [inline app/routes/circulars.moderation._index.tsx:89]:
```suggestion
            Delete Selected
```

File: app/routes/circulars.moderation._index.tsx [hunk 2]
```diff
@@ -31,45 +39,100 @@ export async function loader({ request }: LoaderFunctionArgs) {
...
+    if (checked) {
+      setSelectedList([...selectedList, id])
+    } else {
+      setSelectedList(selectedList.filter((x) => x !== id))
+    }
+  }
+
+  function setBool() {
+    setDisableSubmit(selectedList.length === 0)
+  }
```
Comment (lpsinger) [inline app/routes/circulars.moderation._index.tsx:76]: You can probably achieve a simpler state update if you just store the _number_ of selected checkboxes: it's initially zero; you increment it whenever a checkbox goes from unchecked to checked; you decrement it whenever a checkbox goes from checked to unchecked.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/circulars.moderation._index.tsx (hunk 2 above)
Comment (dakota002) [reply app/routes/circulars.moderation._index.tsx]: Good call, that worked much better and cleaned up the code a bit too!

File: app/routes/circulars.moderation._index.tsx [hunk 3]
```diff
@@ -60,23 +60,16 @@ export async function action({ request }: ActionFunctionArgs) {
...
 
-  function checkboxOnChange(id: string, checked: boolean) {
+  function checkboxOnChange(checked: boolean) {
     if (checked) {
-      setSelectedList([...selectedList, id])
+      setSelectedCount(selectedCount + 1)
     } else {
-      setSelectedList(selectedList.filter((x) => x !== id))
+      setSelectedCount(selectedCount - 1)
     }
```
Comment (lpsinger) [inline app/routes/circulars.moderation._index.tsx:70]: I don't know if Typescript will allow, but due to JavaScript's rules for type promotion, you could do

```ts
setSelectedCount(selectedCount + checked)
```

Elegantly concise? Unholy terror from beyond? You decide.

Also, is there any benefit to using the callback version of setState here? I confess that I do not totally understand when you need that and when you don't.

```ts
setSelectedCount((old) => old + 1)
```

**Review by lpsinger — APPROVED** [review]

**Pull request merged**